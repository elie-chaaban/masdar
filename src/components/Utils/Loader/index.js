import React from 'react';
import {Row} from 'react-bootstrap';
import {Animate} from '../';
import PropTypes from 'prop-types';
import './style.scss';

const Loader = ({color = '#fff', height = '100%', children, margin = 20}) => {
  return (
    <Row className="justify-content-md-center align-items-center" style={{color, margin, height, minHeight: height, zIndex: 10000000}}>
      <Animate options={{duration: 500}} className="text-center">
        <div className="lds-ellipsis">
          <div style={{backgroundColor: color}}></div>
          <div style={{backgroundColor: color}}></div>
          <div style={{backgroundColor: color}}></div>
          <div style={{backgroundColor: color}}></div>
        </div>
        {children}
      </Animate>
    </Row>
  );
};

Loader.propTypes = {
  color: PropTypes.string,
  height: PropTypes.string,
  children: PropTypes.node
};
export default Loader;
