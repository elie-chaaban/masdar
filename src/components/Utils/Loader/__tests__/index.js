import React from 'react';
import {mount} from 'enzyme';
import Loader from '../';

describe('Loader', () => {
  it('sould have default props', () => {
    const dom = mount(
      <Loader>
        <p>loading...</p>
      </Loader>
    );
    expect(dom.find('.lds-ellipsis').childAt(0).prop('style')).toEqual({backgroundColor: '#fff'});
    expect(dom.find('p')).toHaveLength(1);
    expect(dom.find('Row').prop('style')).toEqual({color: '#fff', margin: 20, height: '100%', minHeight: '100%'});
    expect(dom).toMatchSnapshot();
  });
  it('sould have default props', () => {
    const dom = mount(
      <Loader color="red" margin={0} height="100px">
        <p>loading...</p>
      </Loader>
    );
    expect(dom.find('.lds-ellipsis').childAt(0).prop('style')).toEqual({backgroundColor: 'red'});
    expect(dom.find('p')).toHaveLength(1);
    expect(dom.find('Row').prop('style')).toEqual({color: 'red', margin: 0, height: '100px', minHeight: '100px'});
  });
});
