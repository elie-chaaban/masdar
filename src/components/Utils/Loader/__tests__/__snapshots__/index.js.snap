// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader sould have default props 1`] = `
<Loader>
  <Row
    className="justify-content-md-center align-items-center"
    noGutters={false}
    style={
      Object {
        "color": "#fff",
        "height": "100%",
        "margin": 20,
        "minHeight": "100%",
      }
    }
  >
    <div
      className="justify-content-md-center align-items-center row"
      style={
        Object {
          "color": "#fff",
          "height": "100%",
          "margin": 20,
          "minHeight": "100%",
        }
      }
    >
      <Animate
        animation="fade"
        className="text-center"
        enter={true}
        onAnimationEnd={null}
        options={
          Object {
            "duration": 500,
          }
        }
        unMountOnExit={false}
      >
        <div
          className="text-center"
          onAnimationEnd={[Function]}
          style={
            Object {
              "animationDelay": "undefinedms",
              "animationDirection": undefined,
              "animationDuration": "500ms",
              "animationFillMode": undefined,
              "animationIterationCount": undefined,
              "animationName": "fadeIn",
              "animationPlayState": undefined,
              "animationTimingFunction": undefined,
            }
          }
        >
          <div
            className="lds-ellipsis"
          >
            <div
              style={
                Object {
                  "backgroundColor": "#fff",
                }
              }
            />
            <div
              style={
                Object {
                  "backgroundColor": "#fff",
                }
              }
            />
            <div
              style={
                Object {
                  "backgroundColor": "#fff",
                }
              }
            />
            <div
              style={
                Object {
                  "backgroundColor": "#fff",
                }
              }
            />
          </div>
          <p>
            loading...
          </p>
        </div>
      </Animate>
    </div>
  </Row>
</Loader>
`;
